inc/t_cose/t_cose_common.h

```diff
@@ -791,6 +791,19 @@ enum t_cose_err_t {
  */
 #define T_COSE_OPT_REQUIRE_KID 0x00001000
 
+/**
+ * WARNING: DO NOT use this option flag without understanding the
+ * security consideration of RFC 9459.
+ * By default, the error \ref T_COSE_ERR_UNSUPPORTED_ENCRYPTION_ALG
+ * is returned if non AEAD content encryption algorithms, such as
+ * AES-CTR and AES-CBC, are used.
+ * The sender and recipient can use them only with this option flag.
+ * The sender MUST use AES-CTR in conjunction with an authentication
+ * and integrity mechanism, such as a digital signature.
+ * The recipient MUST validate data authenticity and integrity.
```

Comment:
I don't like using "MUST" here because we have no enforcement mechanism and it might be safe in some use cases. It's an overreach to say this.

The real danger is using ANY public key encryption without an authentication & integrity layer. AEAD is never an alternative to COSE_Sign or such. If we are warning about lack  authentication and integrity is should for any/all of COSE_Encrypt. I find the emphasis on AEAD kind of misplaced relative to COSE_Encrypt without COSE_Sign.

Part of the reason I prefer the warning only in one place in t_cose is so one day we might write more about this and only have to do it in one place.

Reply:
Removed this paragraph, and remains a TODO.
https://github.com/laurencelundblade/t_cose/blob/904484717ffd8d89eaf8a7352544ab2b1bd07618/inc/t_cose/t_cose_common.h#L798-L811